File: A1-WiSe1920Counter/step_counter.rb
require_relative  "base_counter"
class StepCounter < BaseCounter

  def initialize(start=0, step=1)
    super(start)
    @step = step
  end

  def inc()
    @count+=@step
    return self
  end

  def num_inc()
    super()/@step
  end

end

bc = StepCounter.new(4,2)
bc.inc().inc().inc()
puts bc
bc.reset()
bc.inc().inc()
puts bc

File: src/a11/StepCounter.java
package a11;

public class StepCounter extends BaseCounter {

    private int start;
    private int count;
    private int step;

    public StepCounter(int start, int step) {
        super(start);
        this.start = start;
        this.count = start;
        this.step = step;
    }

    public StepCounter() {
        this(0, 1);
    }

    @Override
    public void inc() {
        count += this.step;
    }

    @Override
    public int numInc() {
        return super.numInc() / this.step;
    }

    public static void main(String[] args) {
        StepCounter sc = new StepCounter(4, 2);
        sc.inc();
        sc.inc();
        sc.inc();
        System.out.println(sc);
    }
}

